lib/teleterm/apiserver/handler/handler_auth.go

```diff
@@ -37,6 +37,11 @@ func (s *Handler) Login(ctx context.Context, req *api.LoginRequest) (*api.EmptyR
 	// added by daemon.Service.ResolveClusterURI.
 	clusterClient.MFAPromptConstructor = nil
 
+	err = s.DaemonService.ClearCachedClientsForRoot(cluster.URI)
```

Comment:
Also, I tested it and it works properly with the cache changes, but the line above sets `MFAPromptConstructor` to nil. This is so that when you attempt to log in, we don't show the regular MFA prompt that gets displayed when e.g. we ask for Kube MFA.

This works only because we do `ResolveCluster`, followed by setting `MFAPromptConstructor` and then calling `cluster.LocalLogin` which uses the same `clusterClient` that gets returned from `ResolveCluster`. If we ever change this, this flow might break.

I should've written an integration test for this scenario when I was working on Kube MFA. The test would check that tshd doesn't send `Relogin` over tshd events during local login with MFA.

Reply:
I started writing a test, but I stuck around setting up `tshdEventsService` and `webauthnLogin`, this is quite complex 💀.